lib/stack_master/parameter_resolvers/latest_ami.rb

```diff
@@ -0,0 +1,25 @@
+module StackMaster
+  module ParameterResolvers
+    class LatestAmi < Resolver
+      array_resolver class_name: 'LatestAmis'
+
+      def initialize(config, stack_definition)
+        @config = config
+        @stack_definition = stack_definition
+        @ami_finder = AmiFinder.new(@stack_definition.region)
+      end
+
+      def resolve(value)
+        case value
+        when String
+          owners = ['self']
+          filters = @ami_finder.build_filters_from_string(value)
+        when Hash
+          owners = Array(value.delete('owner_id').to_s || 'self')
+          filters = @ami_finder.build_filters_from_hash(value)
+        end
```

Comment:
Yeah I thought that was the reason for wanting to move to a hash... The way it is now could just as easily be done with a key value string like our other resolver inputs. We'd just need to parse the key value string into a hash first, the rest is the same logic.